server_spidercast/src/MCC_RUM/src/RumRxTxApi.c

```diff
@@ -126,10 +126,13 @@ static void rumImutex_unlock(void)
 
 static int rumInstanceCounter = 1;
 static char* createRUMInstanceName(void){
-  char  result[64];
-  snprintf(result, sizeof(result), "RUM.%d", rumInstanceCounter);
+  char* result = malloc(64 * sizeof(char)); //to dynamically allocate 64 bytes 
+  if (result == NULL){
+    return NULL;
+  }
+  snprintf(result, 64, "RUM.%d", rumInstanceCounter);
   rumInstanceCounter++;
-  return rmm_strdup(result);
```

Comment:
Can you explain what you are trying to acheive with this change? In both cases, the memory will be allocated on the heap (previously by strdup now by malloc. Now there's an extra malloc but it doesn't obviously fix any problem to me (and the comment doesn't look right - you can see that it's allocating 64 bytes from the code - a comment there should explain why 64.

Reply:
The issue in this was that llmRegisterTraceComponent is trying to read 64 bytes from string that might be shorter than 64 bytes, which is leading to the warning.
I was trying with the approach that the memory that's returned should be allocated and initialized.

Reply:
Can you share the warning?

Reply:
Sure , this is the warning,
`[exec] /opt/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c:137:5: warning: llmRegisterTraceComponent reading 64 bytes from a region of size 8 [-Wstringop-overread]
[exec] /opt/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c:138:5: warning: llmRegisterTraceComponent reading 64 bytes from a region of size 7 [-Wstringop-overread]
[exec] /opt/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c:139:5: warning: llmRegisterTraceComponent reading 64 bytes from a region of size 7 [-Wstringop-overread]`

Reply:
So the change changes the allocate to be 64 bytes as we currently read 64 bytes. Might it be better to change the usage of it to only read up to the null (I've not looked yet).

Reply:
Hmm.... If I look at the error I see:
```
/home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c: In function ‘registerRUMTraceComponents’:
/home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c:137:5: warning: ‘llmRegisterTraceComponent’ reading 64 bytes from a region of size 8 [-Wstringop-overread]
  137 |     llmRegisterTraceComponent(instanceName,300,"RUM_API",&(tcHandles[0]),NULL);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jon/work/amlen/repos/amlen/server_spidercast/src/MCC_RUM/src/RumRxTxApi.c:137:5: note: referencing argument 3 of type ‘const char[64]’
```
Note: the last bit of that warning says it's about argument 3 ("RUM_API") not the instance name.

Reply:
(So I think the change won't fix the warning - 
I think probably:
```
LLM_API int	llmRegisterTraceComponent(const char* instanceName, int componentId, const COMPONENT_NAME componentName,
```
needs to change the type of componentName to `const char *componentName`

Reply:
yes, that can be done. I have already thought about that approach too. I'll make that code change and commit it

Reply:
Instead of assuming a fixed size, we can calculate the actual string length using strlen(instanceName) to get number of characters before null terminator. 
Which will pass only the useful part of the string (up to the null terminator)

Reply:
(and added a static keyword... I think this bit of code can just be reverted to how it was

Reply:
oh, I'll get this part of the code reverted back to the original one.

Reply:
actually that was the part of the code change that I had done. But after the latest code changes this particular change of (strdup) is now not required. That was a miss from my end, I'll revert back the code to original one.